services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceTest.kt

```diff
@@ -656,15 +659,33 @@ class HakemusServiceTest {
             every { hakemusRepository.save(any()) } answers { firstArg() }
             every { alluStatusRepository.getReferenceById(1) } returns AlluStatus(1, updateTime)
             every { alluStatusRepository.save(any()) } answers { firstArg() }
-            justRun { paatosService.saveKaivuilmoituksenPaatos(any(), any()) }
+            justRun {
+                when (applicationStatus) {
+                    ApplicationStatus.DECISION -> {
+                        paatosService.saveKaivuilmoituksenPaatos(any(), any())
+                    }
+                    ApplicationStatus.OPERATIONAL_CONDITION -> {
+                        paatosService.saveKaivuilmoituksenToiminnallinenKunto(any(), any())
+                    }
+                    else -> throw IllegalArgumentException("Invalid status")
+                }
+            }
```

Comment:
When possible, complex reasoning should be avoided inside the stubbing methods. A method reference can be stored in a value and called inside the stubbing block.

This method is only called with the statuses listed, so there's no reason to account for other cases.
```suggestion
            val saveMethod =
                when (applicationStatus) {
                    ApplicationStatus.DECISION -> paatosService::saveKaivuilmoituksenPaatos
                    else -> paatosService::saveKaivuilmoituksenToiminnallinenKunto
                }
            justRun { saveMethod(any(), any()) }
```

The same stored method can be used for verifying.

Reply:
In fact, I think the `mockAlluDownload` and `verifyAlluDownload` methods and their uses could be simplified in `HakemusServiceITest` by using method references.

Reply:
I changed these to use method references.